### PR TITLE
Add 2023 Massachusetts Tax Rules

### LINKED ISSUE #4180: Add 2023 Massachusetts Tax Rules
2023 Income Tax Form Instructions: https://www.mass.gov/doc/2023-form-1-instructions/download
(Child and Dependent Tax Credit and EITC rate have already been updated)

### PR DESCRIPTION
**Pull request by Mshuning**
Fixes #4180

### CODE REVIEW

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py [hunk 1]
```diff
@@ -13,9 +13,12 @@ class ma_dependent_or_dependent_care_credit(Variable):
     defined_for = StateCode.MA
 
     def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ma.tax.income.credits.dependent_care
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py:16]: Is the child and family tax credit a completely new credit concept or did MA just merge the two previous credits? I dont think I see the exact definition in the 2021 tax forms

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]:
```suggestion
- bump: minor
```

File: policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py [hunk 3]
```diff
@@ -13,9 +13,12 @@ class ma_dependent_or_dependent_care_credit(Variable):
...
-        return max_(
-            tax_unit("ma_dependent_credit", period),
-            tax_unit("ma_dependent_care_credit", period),
-        )
+        if p.in_effect:
+            return max_(
+                tax_unit("ma_dependent_credit", period),
+                tax_unit("ma_dependent_care_credit", period),
+            )
+        return tax_unit("ma_dependent_credit", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py:24]:
```suggestion
        dependent_credit = tax_unit("ma_dependent_credit", period)
        if p.in_effect:
            return max_(
                dependent_credit,
                tax_unit("ma_dependent_care_credit", period),
            )
        return dependent_credit
```

File: policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py:16]: The question is whether the rules stay the same or if we need to re-write the logic

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.yaml [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
   period: 2022
-  absolute_error_margin: 0
   input:
     ma_dependent_credit: 1
     ma_dependent_care_credit: 2
     state_code: MA
   output:
     ma_dependent_or_dependent_care_credit: 2
+
+- name: 2023 Max of the credits.
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.yaml:10]: add 2024 test

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent_care/in_effect.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+description: Massachusetts replaces the Dependent Care Tax Credit with the Child and Family Tax Credit if this is true.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent_care/in_effect.yaml:1]:
```suggestion
description: Massachusetts replaces the Dependent and Dependent Care Tax Credit with the Child and Family Tax Credit if this is false.  
```

File: policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py [hunk 6]
```diff
@@ -13,9 +13,12 @@ class ma_dependent_or_dependent_care_credit(Variable):
     defined_for = StateCode.MA
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.py:13]: add documentation metadata stating that from 2023 this is called the Child and Family tax credit

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/tests/policy/baseline/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.yaml [hunk 7]
```diff
@@ -1,9 +1,17 @@
-- name: Max of the credits.
+- name: 2022 Max of the credits.
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ma/tax/income/credits/ma_dependent_or_dependent_care_credit.yaml:1]:
```suggestion
- name: 2022 max of the credits.
```

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/senior_circuit_breaker/income/disallowed_deductions.yaml [hunk 8]
```diff
@@ -13,7 +13,7 @@ values:
 metadata:
   unit: list
   name: ma_scb_disallowed_deductions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/senior_circuit_breaker/income/disallowed_deductions.yaml:15]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ma/tax/income/credits/senior_circuit_breaker/income/disallowed_deductions.yaml:15]: Could you remove all -name tags if thee are more while we are in here

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Pull request merged**